tests/e2e/multi_vc_preferential_topology.go

```diff
@@ -0,0 +1,776 @@
+
+	ginkgo.AfterEach(func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		ginkgo.By(fmt.Sprintf("Deleting all statefulsets in namespace: %v", namespace))
+		fss.DeleteAllStatefulSets(client, namespace)
+		ginkgo.By(fmt.Sprintf("Deleting service nginx in namespace: %v", namespace))
+		err := client.CoreV1().Services(namespace).Delete(ctx, servicename, *metav1.NewDeleteOptions(0))
+		if !apierrors.IsNotFound(err) {
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+		framework.Logf("Perform preferred datastore tags cleanup after test completion")
+		err = deleteTagCreatedForPreferredDatastore(masterIp, sshClientConfig, allowedTopologyRacks,
+			true)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		framework.Logf("Recreate preferred datastore tags post cleanup")
+		err = createTagForPreferredDatastore(masterIp, sshClientConfig, allowedTopologyRacks, true)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	})
+
+	/* Testcase-1:
+		Add preferential tag in all the Availability zone's of VC1 and VC2  → change the preference during
+		execution
+
+	    1. Create SC default parameters without any topology requirement.
+	    2. In each availability zone for any one datastore add preferential tag in VC1 and VC2
+	    3. Create 3 statefulset with 10 replica's
+	    4. Wait for all the  PVC to bound and pods to reach running state
+	    5. Verify that since the preferred datastore is available, Volume should get created on the datastores
+		 which has the preferencce set
+	    6. Make sure common validation points are met on PV,PVC and POD
+	    7. Change the Preference in any 2 datastores
+	    8. Scale up the statefulset to 15 replica
+	    9. The volumes should get provision on the datastores which has the preference
+	    10. Clear the data
+	*/
+	ginkgo.It("Tag one datastore as preferred each in VC1 and VC2 and verify it is honored", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		parallelStatefulSetCreation = true
+		preferredDatastoreChosen = 1
+		preferredDatastorePaths = nil
+		sts_count := 3
+		stsReplicas = 10
+		var dsUrls []string
+		scaleUpReplicaCount = 15
+		stsScaleDown = false
+
+		ginkgo.By("Create SC with default parameters")
+		scSpec := getVSphereStorageClassSpec(defaultNginxStorageClassName, nil, nil, "",
+			"", false)
+		sc, err := client.StorageV1().StorageClasses().Create(ctx, scSpec, metav1.CreateOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			err = client.StorageV1().StorageClasses().Delete(ctx, sc.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		// choose preferred datastore
+		ginkgo.By("Tag preferred datastore for volume provisioning in VC1 and VC2")
+		for i := 0; i < 2; i++ {
+			paths, err := tagPreferredDatastore(masterIp, sshClientConfig, allowedTopologies[0].Values[i],
+				preferredDatastoreChosen, ClusterdatastoreListVC[i], nil, true, i)
```

Comment:
Can you please explain what's the logic for tagging preferred datastores in VC1 and VC2.